packages/wonder-blocks-search-field/src/components/search-field.js

```diff
@@ -0,0 +1,257 @@
+import {View} from "@khanacademy/wonder-blocks-core";
+import IconButton from "@khanacademy/wonder-blocks-icon-button";
+import {TextField} from "@khanacademy/wonder-blocks-form";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import type {StyleType, AriaProps} from "@khanacademy/wonder-blocks-core";
+
+import {defaultLabels} from "../util/constants.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /**
+     * ARIA label for the clear button. Defaults to "Clear search".
+     */
+    clearAriaLabel?: string,
+
+    /**
+     * The unique identifier for the input.
+     */
+    id: string,
+
+    /**
+     * The text input value.
+     */
+    value: string,
+
+    /**
+     * Provide hints or examples of what to enter. This shows up as
+     * a grayed out text in the field before a value is entered.
+     */
+    placeholder?: string,
+
+    /**
+     * Makes a read-only input field that cannot be focused.
+     * Defaults to false.
+     */
+    disabled?: boolean,
+
+    /**
+     * Changes the default focus ring color to fit a dark background.
+     */
+    light?: boolean,
+
+    /**
+     * Custom styles for the main wrapper.
+     */
+    style?: StyleType,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Called when the value has changed.
+     */
+    onChange: (newValue: string) => mixed,
+
+    /**
+     * Handler that is triggered when this component is clicked. For example,
+     * use this to adjust focus in parent component. This gets called when we
+     * click the dismiss icon button within the SearchField.
+     */
+    onClick?: () => mixed,
+
+    /**
+     * Called when a key is pressed.
+     */
+    onKeyDown?: (event: SyntheticKeyboardEvent<HTMLInputElement>) => mixed,
+
+    /**
+     * Called when the element has been focused.
+     */
+    onFocus?: (event: SyntheticFocusEvent<HTMLInputElement>) => mixed,
+
+    /**
+     * Called when the element has been blurred.
+     */
+    onBlur?: (event: SyntheticFocusEvent<HTMLInputElement>) => mixed,
+|};
+
+type DefaultProps = {|
+    clearAriaLabel: $PropertyType<Props, "clearAriaLabel">,
+    disabled: $PropertyType<Props, "disabled">,
+    light: $PropertyType<Props, "light">,
+|};
+
+type State = {|
+    /**
+     * The user focuses on this field.
+     */
+    focused: boolean,
+|};
+
+export default class SearchField extends React.Component<Props, State> {
+    static defaultProps: DefaultProps = {
+        clearAriaLabel: defaultLabels.clearSearch,
+        disabled: false,
+        light: false,
+    };
+
+    state: State = {
+        focused: false,
+    };
+
+    handleDismiss: () => void = () => {
+        const {onClick, onChange} = this.props;
+        // Empty the search text and focus the SearchField
+        onChange("");
+        if (onClick) {
+            onClick();
```

Comment:
I took the `handleDismiss` code directly from `search-text-input.js`, but I'm a bit confused about why it calls `onClick` in this function. I feel like `onClick` should be reserved for clicking into the text field? Should this be a different prop named `onClear` instead?

Reply:
I think `onClick` _is_ passed into the render function, so it's kind of weird that it's also here? What do yall think?

Reply:
The description for the prop says the following

```
/**
     * Handler that is triggered when this component is clicked. For example,
     * use this to adjust focus in parent component. This gets called when we
     * click the dismiss icon button within the SearchField.
     */
    onClick?: () => mixed,
    ```
    
Should `onClick` not be passed into the render function `View`? Should it only be used on the dismiss button? I feel like that's a bit unintuitive.

Reply:
Juan and I have responded in the corresponding Slack thread. 🙂